Marlin/src/module/probe.cpp

```diff
@@ -691,13 +692,15 @@ bool Probe::probe_down_to_z(const_float_t z, const_feedRate_t fr_mm_s) {
 float Probe::run_z_probe(const bool sanity_check/*=true*/) {
   DEBUG_SECTION(log_probe, "Probe::run_z_probe", DEBUGGING(LEVELING));
 
+  const float zoffs = SUM_TERN(HAS_HOTEND_OFFSET, -offset.z, hotend_offset[active_extruder].z);
```

Comment:
It's unclear whether the hotend offset in Z should always have bearing on the probe height, even when the nozzle is used as the probe. The hotend offset is applied to the logical position as an element of tool change, for a start, and may or may not result in a Z move. There isn't a well-defined standard for associating a logical coordinate alteration with the native/machine position w/r/t a probe. To help alleviate confusion on these points, it would help to have some use cases and examples, describing the procedure. Of course, the logs from `DEBUG_LEVELING_FEATURE` plus `M111 S8` should provide good concrete examples. We might try using the Simulator if multi-tool machines are hard to access.

Reply:
Maybe just make it optional, e.g. APPLY_TOOL_OFFSETS_TO_PROBE? My initial thought was, if different tool offsets do not apply to probe movement, stick with your default tool for probing (which is the current case, anyway).